src/uShip.Logging/LogBuilders/PropertiesDictionaryExtensions.cs

```diff
@@ -11,14 +14,49 @@ private static string Sanitize(string input)
             return input.IfNotNull(x => x.SanitizeSensitiveInfo().RemoveViewState());
         }
 
+        public static IDictionary NewSanitizedDictionary(this IDictionary data)
+        {
+            if (data == null)
+            {
+                return null;
+            }
+
+            IDictionary sanitizedData = new Dictionary<String, object>();
+
+            foreach (var key in data.Keys)
+            {
+                if (data[key] == null)
+                {
+                    continue;
+                }
+
+                if (data[key] is string && LogDataSanitizeExtensions.SensitiveFieldNames.Contains(key))
```

Comment:
I don't think this does what we are expecting. This is saying, if the key is "password", then set dict["password"] = ****.

That's not really what we want.  In our case,  the dictionary contains another object or string, and it's VALUE has "password" in it.